### PR TITLE
Lgirard/feat/tdp 10591 tql dsel convertor

### PR DESCRIPTION
**Pull request by laurentgir**
**What is the problem this Pull Request is trying to solve?**
 
 see TDP-10591
 
**What is the chosen solution to this problem?**
 
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-tql/daikon-tql-dsel/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>org.talend.maplang</groupId>
+            <artifactId>org.talend.maplang.el.interpreter</artifactId>
+            <version>${maplang.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.talend.maplang</groupId>
+            <artifactId>org.talend.maplang.el.parser</artifactId>
+            <version>1.4.0</version>
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/pom.xml:45]:
```suggestion
            <version>${maplang.version}</version>
```

File: daikon-tql/daikon-tql-dsel/pom.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        <dependency>
+            <groupId>org.talend.daikon</groupId>
+            <artifactId>daikon-tql-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/pom.xml:56]: why using junit 4 ? Can you migrate to junit 5 please ?

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import org.talend.maplang.el.parser.model.ELNode;
+import org.talend.maplang.el.parser.model.ELNodeType;
+import org.talend.tql.excp.TqlException;
+import org.talend.tqldsel.DSELConverter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.talend.tqldsel.DSELConverter.wrapNode;
+
+public class DSELVisitorFunctionCallTest {
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:11]: why using junit 4 ? Can you migrate to junit 5 please ?

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    @Test
+    public void testParseIsNull() {
+        final String query = "field1 is null";
+        ELNode actual = DSELConverter.convert(query);
+        ELNode nullNode = buildNode(ELNodeType.FUNCTION_CALL, "isNull", "field1", "null");
+
+        System.out.println("\n\n---> TQL Expression  : " + query);
+        System.out.println("---> DSEL Expression : " + actual.toString());
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:20]: to delete

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    @Test
+    public void testParseIsEmtpy() {
+        final String query = "field1 is empty";
+        ELNode actual = DSELConverter.convert(query);
+        ELNode isEmptyNode = buildNode(ELNodeType.FUNCTION_CALL, "isEmpty", "field1", "empty");
+
+        System.out.println("\n\n---> TQL Expression  : " + query);
+        System.out.println("---> DSEL Expression : " + actual.toString());
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:44]: to delete

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    public void testParseRegEx() {
+        final String query = "name ~ '^[A-Z][a-z]*$'";
+        ELNode actual = DSELConverter.convert(query);
+
+        ELNode regexNode = new ELNode(ELNodeType.FUNCTION_CALL, "matches");
+        regexNode.addChild(new ELNode(ELNodeType.HPATH, "name"));
+        regexNode.addChild(new ELNode(ELNodeType.STRING_LITERAL, "^[A-Z][a-z]*$"));
+
+        System.out.println("\n\n---> TQL Expression  : " + query);
+        System.out.println("---> DSEL Expression : " + actual.toString());
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:72]: to delete

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorNotTest.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import org.junit.Test;
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorNotTest.java:1]: migrate to junit 5

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorOrAndTest.java [hunk 8]
```diff
@@ -0,0 +1,99 @@
+import org.junit.Test;
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorOrAndTest.java:1]: migrate to junit 5

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorSimpleComparisonTest.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
+import org.junit.Test;
```
Comment (fabballe) [inline daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorSimpleComparisonTest.java:1]: migrate to junit 5

File: daikon-tql/daikon-tql-dsel/pom.xml (hunk 2 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/pom.xml]: done

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java (hunk 3 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java:11]: done

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java (hunk 4 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java]: done

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java (hunk 5 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java]: done

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java (hunk 6 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorFunctionCallTest.java]: done

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorOrAndTest.java (hunk 8 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorOrAndTest.java]: done

File: daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorSimpleComparisonTest.java (hunk 9 above)
Comment (laurentgir) [reply daikon-tql/daikon-tql-dsel/src/test/java/DSELVisitorSimpleComparisonTest.java]: done

**Comment by laurentgir** [comment]
Closing this PR, made on a fork

**Pull request closed without merging**